pytorch_lightning/plugins/environments/kubeflow_environment.py

```diff
@@ -47,6 +50,14 @@ def main_address(self) -> str:
     def main_port(self) -> int:
         return int(os.environ["MASTER_PORT"])
 
+    @staticmethod
+    def detect() -> bool:
+        """Returns ``True`` if the current process was launched using Kubeflow PyTorchJob."""
+        required_env_vars = ("KUBERNETES_PORT", "MASTER_ADDR", "MASTER_PORT", "WORLD_SIZE", "RANK")
+        # torchelastic sets these. Make sure we're not in torchelastic
+        excluded_env_vars = ("GROUP_RANK", "LOCAL_RANK", "LOCAL_WORLD_SIZE")
+        return all(v in os.environ for v in required_env_vars) and not any(v in os.environ for v in excluded_env_vars)
```

Comment:
we can be cool here with sets

```suggestion
        required_env_vars = {"KUBERNETES_PORT", "MASTER_ADDR", "MASTER_PORT", "WORLD_SIZE", "RANK"}
        # torchelastic sets these. Make sure we're not in torchelastic
        excluded_env_vars = {"GROUP_RANK", "LOCAL_RANK", "LOCAL_WORLD_SIZE"}
        env_vars = os.environ.keys()
        return env_vars == required_env_vars and env_vars.isdisjoint(excluded_env_vars)
```

Reply:
that seems all good, but should we really do it now? I meant to only move things around here and not change at the same time.

Reply:
why not? tests should catch any issues with the change (assuming we have tests for this)

Reply:
it's not about correctness of the implementation.
I don't feel comfortable moving code and changing it at the same time because of the review process.

Reply:
Let's make this change in another PR then.